pkg/parse/parser_test.go

```diff
@@ -8,9 +8,84 @@ import (
 )
 
 func TestReadYAMLWriteJSON(t *testing.T) {
+	// XXX: this will fail with 1.24 which should clean up the output via omitzero
 	in := "name: test\n"
-	want := "{\n  \"name\": \"test\"\n}"
+	want := `{
+  "accounts": {
+    "groups": null,
+    "users": null
+  },
+  "dnf": {},
+  "fips": {},
+  "ignition": null,
+  "installer": {
+    "anaconda": {},
+    "coreos": {}
+  },
+  "kernel": {},
+  "locale": {},
+  "name": "test",
+  "network": {
+    "firewall": {}
+  },
+  "openscap": {
+    "profile_id": ""
+  },
+  "registration": {
+    "fdo": {
+      "manufacturing_server_url": ""
+    },
+    "redhat": {
+      "connector": {
+        "enabled": false
+      },
+      "insights": {
+        "enabled": false
+      },
+      "subscription_manager": {}
+    }
+  },
+  "storage": {
+    "partitions": null,
+    "type": ""
+  },
+  "systemd": {},
+  "timedate": {}
+}`
 
+	// XXX: however storing back YAML will not respect omitzero and this
+	// needs to be fixed (relevant only for the convertor)
+	want2 := `accounts:
+  groups: null
+  users: null
+dnf: {}
+fips: {}
+ignition: null
+installer:
+  anaconda: {}
+  coreos: {}
+kernel: {}
+locale: {}
+name: test
+network:
+  firewall: {}
+openscap:
+  profile_id: ""
+registration:
+  fdo:
+    manufacturing_server_url: ""
+  redhat:
+    connector:
+      enabled: false
+    insights:
+      enabled: false
+    subscription_manager: {}
+storage:
+  partitions: null
+  type: ""
+systemd: {}
+timedate: {}
+`
```

Comment:
YAML parser will not respect `omitzero` this needs to be solved separately. Even if image builder does not need this technically, I want the convertor to generate a nice output.